Really appreciate the PR—these are good fixes and tough to catch!

Glad to have the brahe implementation brought into agreement with others of the the model. It's based on an old implementation of mine in Julia, retranslated, and I expected that I just never realized originally it was supposed to use the previous day.

The CSSI loader is just a pure bug 🤦🏻‍♂️.

There are two failing python unit tests coming from the change of the file. These should be fixed before merging. I can get to it later this week, but if you're able to do it sooner, I'll hit merge as soon as it lands.

We can ignore the failing `test-examples` / `test-rust` those are flakey integration tests against the external data dependencies, which have been getting worse of late (todo list item of mine to fix).

```
=================================== FAILURES ===================================
_________________________ test_get_global_known_values _________________________

sw_test_filepath = '/home/runner/work/brahe/brahe/test_assets/sw19571001.txt'

    def test_get_global_known_values(sw_test_filepath):
        """Test getting known values from global provider using test file"""
        sw = brahe.FileSpaceWeatherProvider.from_file(sw_test_filepath, "Hold")
        brahe.set_global_space_weather_provider(sw)
    
        # MJD 36112.0 = 1957-10-01
        mjd = 36112.0
    
        # Known Kp value at 00:00 should be 4+1/3 (from raw value 43)
        assert brahe.get_global_kp(mjd) == pytest.approx(4.0 + 1.0 / 3.0, abs=1e-10)
    
        # Known Ap daily average: 21
        assert brahe.get_global_ap_daily(mjd) == pytest.approx(21.0, abs=1e-10)
    
        # Known F10.7 adjusted: 269.8
>       assert brahe.get_global_f107_observed(mjd) == pytest.approx(269.8, abs=1e-10)
E       assert 269.3 == 269.8 ± 1.0e-10
E         
E         comparison failed
E         Obtained: 269.3
E         Expected: 269.8 ± 1.0e-10

tests/space_weather/test_global_sw.py:33: AssertionError
_________________________ test_from_file_known_values __________________________

sw_test_filepath = '/home/runner/work/brahe/brahe/test_assets/sw19571001.txt'

    def test_from_file_known_values(sw_test_filepath):
        """Test that known values from 1957-10-01 are read correctly"""
        sw = brahe.FileSpaceWeatherProvider.from_file(sw_test_filepath, "Hold")
    
        # MJD 36112.0 = 1957-10-01
        mjd = 36112.0
    
        # Known Kp values from first data line: 43 40 30 20 37 23 43 37
        # These are stored as (integer * 10 + fraction), where fraction 3=+1/3, 7=+2/3
        kp_all = sw.get_kp_all(mjd)
        assert len(kp_all) == 8
        assert kp_all[0] == pytest.approx(4.0 + 1.0 / 3.0, abs=1e-10)  # 43 -> 4+1/3
        assert kp_all[1] == pytest.approx(4.0, abs=1e-10)  # 40 -> 4.0
        assert kp_all[2] == pytest.approx(3.0, abs=1e-10)  # 30 -> 3.0
        assert kp_all[3] == pytest.approx(2.0, abs=1e-10)  # 20 -> 2.0
        assert kp_all[4] == pytest.approx(3.0 + 2.0 / 3.0, abs=1e-10)  # 37 -> 3+2/3
        assert kp_all[5] == pytest.approx(2.0 + 1.0 / 3.0, abs=1e-10)  # 23 -> 2+1/3
        assert kp_all[6] == pytest.approx(4.0 + 1.0 / 3.0, abs=1e-10)  # 43 -> 4+1/3
        assert kp_all[7] == pytest.approx(3.0 + 2.0 / 3.0, abs=1e-10)  # 37 -> 3+2/3
    
        # Known Ap values: 32 27 15 7 22 9 32 22
        ap_all = sw.get_ap_all(mjd)
        assert len(ap_all) == 8
        assert ap_all[0] == pytest.approx(32.0, abs=1e-10)
        assert ap_all[1] == pytest.approx(27.0, abs=1e-10)
        assert ap_all[2] == pytest.approx(15.0, abs=1e-10)
        assert ap_all[3] == pytest.approx(7.0, abs=1e-10)
        assert ap_all[4] == pytest.approx(22.0, abs=1e-10)
        assert ap_all[5] == pytest.approx(9.0, abs=1e-10)
        assert ap_all[6] == pytest.approx(32.0, abs=1e-10)
        assert ap_all[7] == pytest.approx(22.0, abs=1e-10)
    
        # Known Ap daily average: 21
        assert sw.get_ap_daily(mjd) == pytest.approx(21.0, abs=1e-10)
    
        # Known F10.7 adjusted: 269.8 (get_f107_observed returns adjusted value)
>       assert sw.get_f107_observed(mjd) == pytest.approx(269.8, abs=1e-10)
E       assert 269.3 == 269.8 ± 1.0e-10
E         
E         comparison failed
E         Obtained: 269.3
E         Expected: 269.8 ± 1.0e-10

tests/space_weather/test_sw_file_provider.py:67: AssertionError
```

--- a/tests/space_weather/test_global_sw.py
+++ b/tests/space_weather/test_global_sw.py
@@ -30,7 +30,10 @@ def test_get_global_known_values(sw_test_filepath):
     assert brahe.get_global_ap_daily(mjd) == pytest.approx(21.0, abs=1e-10)
 
     # Known F10.7 adjusted: 269.8
-    assert brahe.get_global_f107_observed(mjd) == pytest.approx(269.8, abs=1e-10)
+    assert brahe.get_global_f107_adjusted(mjd) == pytest.approx(269.8, abs=1e-10)
+
+    # Known F10.7 observed: 269.8
+    assert brahe.get_global_f107_observed(mjd) == pytest.approx(269.3, abs=1e-10)
 
     # Known International Sunspot Number: 334
     assert brahe.get_global_sunspot_number(mjd) == 334

--- a/tests/space_weather/test_sw_file_provider.py
+++ b/tests/space_weather/test_sw_file_provider.py
@@ -63,8 +63,11 @@ def test_from_file_known_values(sw_test_filepath):
     # Known Ap daily average: 21
     assert sw.get_ap_daily(mjd) == pytest.approx(21.0, abs=1e-10)
 
-    # Known F10.7 adjusted: 269.8 (get_f107_observed returns adjusted value)
-    assert sw.get_f107_observed(mjd) == pytest.approx(269.8, abs=1e-10)
+    # Known F10.7 adjusted: 269.8
+    assert sw.get_f107_adjusted(mjd) == pytest.approx(269.8, abs=1e-10)
+
+    # Known F10.7 observed: 269.3
+    assert sw.get_f107_observed(mjd) == pytest.approx(269.3, abs=1e-10)
 
     # Known International Sunspot Number: 334
     assert sw.get_sunspot_number(mjd) == 334

